lib/_http_incoming.js

```diff
@@ -121,7 +121,7 @@ IncomingMessage.prototype._addHeaderLines = function(headers, n) {
 // and drop the second. Extended header fields (those beginning with 'x-') are
 // always joined.
 IncomingMessage.prototype._addHeaderLine = function(field, value, dest) {
-  field = field.toLowerCase();
+  field = field.toLowerCase().trim();
```

Comment:
@jasnell why is it hard to do it in http-parser? It seems to me that it is just a matter of adding two extra states.